tests/rptest/archival/s3_client.py

```diff
@@ -292,3 +295,10 @@ def list_objects(self, bucket):
                                            Key=item['Key'],
                                            ETag=item['ETag'][1:-1],
                                            ContentLength=item['Size'])
+
+    def list_buckets(self):
```

Comment:
nit: we should start adding type annotations to new code. Ping me and I'll help you get your environment set up.